server/app/services/applicant/ApplicantService.java

```diff
@@ -519,41 +585,125 @@ private void notifyTiSubmitter(
+        .thenRunAsync(
+            () -> {
+              // Not blocking since these are already completed
+              Optional<String> email = emailFuture.join();
+              Locale locale = localeFuture.join().orElse(LocalizedStrings.DEFAULT_LOCALE);
+              if (email.isEmpty()) {
+                return;
+              }
+              String civiformLink = baseUrl;
+              Messages messages =
+                  messagesApi.preferred(ImmutableSet.of(Lang.forCode(locale.toLanguageTag())));
+              String subject =
+                  messages.at(
+                      MessageKey.EMAIL_APPLICATION_RECEIVED_SUBJECT.getKeyName(), programName);
+              String message =
+                  String.format(
+                      "%s\n%s",
+                      messages.at(
+                          MessageKey.EMAIL_APPLICATION_RECEIVED_BODY.getKeyName(),
+                          programName,
+                          applicantId,
+                          applicationId),
+                      messages.at(MessageKey.EMAIL_LOGIN_TO_CIVIFORM.getKeyName(), civiformLink));
+              if (isStaging) {
+                amazonSESClient.send(stagingApplicantNotificationMailingList, subject, message);
+              } else {
+                amazonSESClient.send(email.get(), subject, message);
+              }
+            },
+            httpExecutionContext.current());
+  }
+
+  /**
+   * Email the applicant with the status' defined email, if the applicant's email address exists and
+   * the status has an email associated with it.
+   *
+   * @param applicantId the ID of the applicant
+   * @param programDef the ProgramDefinition that the applicant applied for
+   * @param status the status from which to get the email body to send
+   */
+  private CompletionStage<Void> maybeNotifyApplicantWithStatus(
+      long applicantId, ProgramDefinition programDef, StatusDefinitions.Status status) {
+    Optional<LocalizedStrings> emailBody = status.localizedEmailBodyText();
+    CompletableFuture<Optional<String>> getEmailFuture =
+        getEmail(applicantId).toCompletableFuture();
+    CompletableFuture<Optional<Locale>> getLocaleFuture =
+        getPreferredLocale(applicantId).toCompletableFuture();
+    return CompletableFuture.allOf(getEmailFuture, getLocaleFuture)
+        .thenRunAsync(
+            () -> {
+              // Not blocking since they've already completed
+              Optional<Locale> maybeLocale = getLocaleFuture.join();
+              Optional<String> maybeEmail = getEmailFuture.join();
+              Locale locale = maybeLocale.orElse(LocalizedStrings.DEFAULT_LOCALE);
+              if (emailBody.isPresent() && maybeEmail.isPresent()) {
+                String programNameForEmail = programDef.localizedName().getOrDefault(locale);
+                // TODO(#3377): Similar to the code in ProgramAdminApplicationService, we are mixing
+                // english
```

Comment:
Thanks for catching this.  I've noticed the formatter does this sometimes.  I wonder if we can make it smarter somehow.

Reply:
Also, this comment is no longer necessary anyway :)

Reply:
@dkatzz As it turns out, I missed that ProgramAdminApplicationService is also sending out emails, which I think handles sending emails when the admin goes and changes the status on an application.  Is it fine to do things synchronously in there, since it's an admin-facing thing, or do we need to do things asynchronously there as well?  Looks like it's all pretty much sync already.

Reply:
Hmm based on [this](https://docs.civiform.us/contributor-guide/developer-guide/development-standards#async-request-handling) in the docs, it seems okay if things are synchronous on the admin side. We may want to consider refactoring that code in the future to be asynchronous though.